### PR TITLE
Updated documentation for reduced op build process.

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Documentation updates for #10030.

**Motivation and Context**
Update documentation.

### CODE REVIEW

File: docs/build/custom.md [hunk 1]
```diff
@@ -34,7 +34,14 @@ The operators that are included are specified at build time, in a [configuration
...
-* NOTE: Building will edit some of the ONNX Runtime source files to exclude unused kernels. If you wish to go back to creating a full build, or wish to change the operator kernels included, you MUST run `git reset --hard` or `git checkout HEAD -- ./onnxruntime/core/providers` from the root directory of your local ONNX Runtime repository to undo these changes.
+
+* NOTE: Building will edit some of the ONNX Runtime source files to exclude unused kernels.
+
+  In particular, this source modification will be done during the "update" build phase, which is enabled by default or explicitly with the `--update` build parameter.
+
+  **ONNX Runtime version 1.10 and earlier:** The source files are modified directly. If you wish to go back to creating a full build, or wish to change the operator kernels included, you MUST run `git reset --hard` or `git checkout HEAD -- ./onnxruntime/core/providers` from the root directory of your local ONNX Runtime repository to undo these changes.
+
+  **ONNX Runtime version 1.11 and later:** Updated versions of the source files are generated in the build directory so there is no need to undo source file changes.
```
Comment (skottmckay) [inline docs/build/custom.md:45]: Should we add that if you do a build with `--clean` you need to make sure any modification is re-applied with the `--update` step?

**Review by natke — APPROVED** [review]
Approve, contingent on Scott being happy :)

File: docs/build/custom.md (hunk 1 above)
Comment (edgchen1) [reply docs/build/custom.md:45]: Good point, I hadn't considered how it interacts with `--clean`. Tried it out and at the moment, it seems that `--clean` (i.e., `cmake --build ... --target clean` will leave the generated files untouched.

There's some indication in the CMake FAQ that the clean target removes files generated by the compiler and linker, which would not include these generated files.
https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#cmake-does-not-generate-a-make-distclean-target-why

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**